### PR TITLE
New MSG_TRACKING_STATE.

### PR DESCRIPTION
**Pull request by wltr**
The previous `MSG_TRACKING_STATE` only supports up to 24 tracking channels. Now it supports up to 63 tracking channels and has a GLONASS FCN field.

(Also, fixes the units of the device monitor message.)

/cc @ljbade @mfine @JoshuaGross @mbavaro @cbeighley

### CODE REVIEW

**Comment by mfine** [comment]
That message id is taken:

```
 - MSG_ACQ_RESULT:
    id: 0x0014
```

otherwise looks great!

**Comment by mfine** [comment]
:shipit: also please generate code for the new messages!

**Comment by wltr** [comment]
@mfine Generated the code as well. Do I also need to bump a version somewhere or is that done by tagging?

File: spec/yaml/swiftnav/sbp/tracking.yaml [hunk 1]
```diff
@@ -250,27 +250,19 @@ definitions:
...
             desc: GNSS signal being tracked
+        - fcn:
+            type: u8
+            desc: Frequency channel number
         - cn0:
-            type: float
-            units: dB Hz
-            desc: Carrier-to-noise density
+            type: u8
+            units: dB Hz / 4
```
Comment (wltr) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: I copied this over from the obs, but shouldn't the unit be `dB Hz * 4`?

@cbeighley @mbavaro

Comment (mbavaro) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: I agree that's confusing but I think the unit is actually quarters of dBHz.. of to write that?

Comment (ljbade) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: 0.25 dB Hz? maybe check the convention used in the other messages

**Comment by mfine** [comment]
Don't need to bump the version - done by tagging.

File: spec/yaml/swiftnav/sbp/tracking.yaml [hunk 2]
```diff
@@ -250,27 +250,19 @@ definitions:
...
-                    values:
-                        - 0: Disabled
-                        - 1: Running
         - sid:
-            type: GnssSignal
+            type: GnssSignal16
             desc: GNSS signal being tracked
+        - fcn:
+            type: u8
+            desc: Frequency channel number
```
Comment (cbeighley) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: @wltr Maybe note that this is GLONASS-specific?

File: c/include/libsbp/tracking.h [hunk 3]
```diff
@@ -79,9 +79,9 @@ signal is in continuous track.
  * measured signal power.
  */
 typedef struct __attribute__((packed)) {
-  u8 state;    /**< Status of tracking channel */
-  sbp_gnss_signal_t sid;      /**< GNSS signal being tracked */
-  float cn0;      /**< Carrier-to-noise density [dB Hz] */
+  gnss_signal16_t sid;    /**< GNSS signal being tracked */
+  u8 fcn;    /**< Frequency channel number */
```
Comment (cbeighley) [inline c/include/libsbp/tracking.h]: Note that this is GLONASS specific?

File: javascript/sbp/tracking.js [hunk 4]
```diff
@@ -129,9 +129,9 @@ MsgTrackingStateDetailed.prototype.fieldSpec.push(['misc_flags', 'writeUInt8', 1
  * power.
  *
  * Fields in the SBP payload (`sbp.payload`):
- * @field state number (unsigned 8-bit int, 1 byte) Status of tracking channel
- * @field sid GnssSignal GNSS signal being tracked
- * @field cn0 number (float, 4 bytes) Carrier-to-noise density
+ * @field sid GnssSignal16 GNSS signal being tracked
+ * @field fcn number (unsigned 8-bit int, 1 byte) Frequency channel number
```
Comment (cbeighley) [inline javascript/sbp/tracking.js]: Note this is GLONASS specific?

File: c/include/libsbp/tracking.h [hunk 5]
```diff
@@ -79,9 +79,9 @@ signal is in continuous track.
...
+  u8 cn0;    /**< Carrier-to-Noise density.  Zero implies invalid cn0. [dB Hz / 4] */
```
Comment (kimvaarala) [inline c/include/libsbp/tracking.h]: I think the final proposal from Colin was to use 0xFF (255) as a invalid value and CN0 should then rail to 254.

Comment (mbavaro) [inline c/include/libsbp/tracking.h]: yes but the "final final" decision was to be consistent with what had been decided for MSG_OBS, thus 0 being invalid!

File: spec/yaml/swiftnav/sbp/tracking.yaml (hunk 1 above)
Comment (wltr) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: @mbavaro Got it. :+1: 
@ljbade Obs message uses dB Hz / 4 too.

File: spec/yaml/swiftnav/sbp/tracking.yaml (hunk 2 above)
Comment (wltr) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: @cbeighley Done.

**Comment by wltr** [comment]
Generated all files. Merging.

**Pull request merged**